server/utils/firebase_algolia_server_code/functions/algoliaFunctions.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Description: code that actually connects to aloglia. 
+ *              `algoliaOperation` parameter is used to enable 
+ *              easy dependency/mock injection.   
+ * Author: Robert Marcus
+ * Date: July 7, 2020
+ */
+
+/**
+ * Creates an algolia record from a firebase entity snapshot.
+ * @param {function(string): !Promise} algoliaOperation a save function
+ * @param {Object} snapshot a json string
+ * @return {!Promise}
+ */
+function createRecordFromEntity(algoliaOperation, snapshot) {
```

Comment:
Why can't all the logic here live inside `index.js`. You could, for example, put these 3 lines for `createRecordFromEntity` directly into the similarly-named function in `index.js:33`. Mocking/DI won't be affected since you still override the default function arguments in the functions over there.

I see that you `require` and `export` this library in `indexConfig` but I can't tell why.

Reply:
Originally done that way, so yes it could all live in 'index.js'. However I wanted the `index.js` code to focus only on things that directly relate to Firebase code and parameterize the Algolia stuff. 

I initialized the `algoliaFunctions` in `indexConfig` for consistency with the other imports.